 overlayfs v10:
 - the standard mount procedure already marks overlayfs
   as a read ony filesystem when the upper filesystem is r/o.
   There is no need to make this, also this confuses
   the options in /proc/mounts for this mounting

Signed-off-by: Jordi Pujol <jordipujolp@gmail.com>

--- a/fs/overlayfs/super.c
+++ b/fs/overlayfs/super.c	2011-08-04 17:01:10.664508474 +0200
@@ -551,10 +551,6 @@ static int ovl_fill_super(struct super_b
 	 */
 	ufs->lower_mnt->mnt_flags |= MNT_READONLY;
 
-	/* If the upper fs is r/o, we mark overlayfs r/o too */
-	if (ufs->upper_mnt->mnt_sb->s_flags & MS_RDONLY)
-		sb->s_flags |= MS_RDONLY;
-
 	if (!(sb->s_flags & MS_RDONLY)) {
 		err = mnt_want_write(ufs->upper_mnt);
 		if (err)
